api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	// Rules are evaluated in order, the first matching rule will be applied,
+	// and the rest will be skipped.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Rules []Rule `json:"rules,omitempty"`
+}
+
+// Rule defines the single authorization rule.
+type Rule struct {
+	// Action defines the action to be taken if the rule matches.
+	Action RuleActionType `json:"action"`
+
+	// Policies contains the list of authorization policies.
+	Policies []Policy `json:"policies"`
```

Comment:
@zhaohuabing with an example, can you explain why we need an extra hierarchy of `policies` and why can't be achieve the same by moving `Subject` and `Permission` under `Rule` ?

Reply:
I was thinking of something like the below:

```
authorization:
  rules:
    - action: "Allow"
      policies:
        - subject: ["Admin"]      
          permission: ["GET", "POST", "DELETE"]
        - subject: ["Viewer"]
          permission: ["Get"]
```

Of course, we can also make it like:

```
authorization:
  rules:
    - action: "Allow"
      subject: ["Admin"]      
      permission: ["GET", "POST", "DELETE"]  
    - action: "Allow"
      subject: ["Viewer"]      
      permission: ["GET", "POST", "DELETE"]  
```

But these would need more RBAC filters to implement it since one RBAC filter can only has one `Action`, either `Allow` or `Deny`.

Reply:
```
But these would need more RBAC filters to implement it since one RBAC filter can only has one Action, either Allow or Deny.
```

I vote to prioritise usability over envoy feasibility here

Reply:
We don't have this problem anymore since we opted to the Matcher API.